### PR TITLE
[Chore] Setup project structure

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/flutter-survey/issues/1

## What happened 👀

☝️ 
 
## Proof Of Work 📹
Screen Shot 2021-04-07 at 14 24 18

### CODE REVIEW

File: lib/configs/app.dart [hunk 1]
```diff
@@ -0,0 +1 @@
+class AppConfigs {}
```
Comment (sleepylee) [inline lib/configs/app.dart:1]: should the file name be `app_config`? 🤔

File: lib/configs/routes.dart [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import 'package:flutter/widgets.dart';
+import 'package:survey/screens/landing/landing_screen.dart';
+
+final Map<String, WidgetBuilder> routes = {
+  "/": (_) => LandingScreen()
+};
```
Comment (sleepylee) [inline lib/configs/routes.dart:6]: curious if we need to add the end of file line or not, maybe you want to run `flutter fmt` to check

File: lib/screens/screen.dart [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    if (Platform.isAndroid) {
+      return Scaffold(
+        body: body,
+      );
+    }
+
+    return CupertinoPageScaffold(
```
Comment (manh-t) [inline lib/screens/screen.dart:20]: In scope of IC, I will assume that we only run project on Android and iOS 😆  I was lazy to open simulator, so I opened project by Chrome and crash happened 🤣

File: lib/core/environment.dart [hunk 4]
```diff
@@ -0,0 +1,14 @@
+class Environment {
+  static const Environment staging = Environment();
```
Comment (manh-t) [inline lib/core/environment.dart:2]: just curious when will you apply flavor to platforms? In other PR? 🙏

Comment (markgravity) [reply lib/core/environment.dart:2]: On it 👍

File: .run/Production.run.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<component name="ProjectRunConfigurationManager">
+  <configuration default="false" name="Production" type="FlutterRunConfigurationType" factoryName="Flutter">
+    <option name="filePath" value="$PROJECT_DIR$/lib/main_production.dart" />
+    <method v="2" />
+  </configuration>
+</component>
```
Comment (sleepylee) [inline .run/Production.run.xml:6]: last one that missed the eof line

**Review by sleepylee — DISMISSED** [review]

**Comment by sleepylee** [comment]
btw please follow the branching convention on the next time, @markgravity 😷 https://nimblehq.co/compass/development/version-control/branch-management/

File: lib/core/environment.dart (hunk 4 above)
Comment (markgravity) [reply lib/core/environment.dart:2]: I will do it in other PR due to building failure in iOS, cause it will take more time to investigate 😢

Comment (markgravity) [reply lib/core/environment.dart]: @manh-t I added it in this PR 😁

File: .run/Production.run.xml (hunk 5 above)
Comment (manh-t) [reply .run/Production.run.xml:7]: you do not fix this? 🐤

File: pubspec.yaml [hunk 6]
```diff
@@ -74,3 +75,28 @@ flutter:
...
+
+flutter_icons:
+  image_path: 'lib/assets/images/app-icon.png'
+  android: true
+  ios: true
+flavorizr:
+  app:
+    android:
+      flavorDimensions: "stage"
+    ios:
```
Comment (manh-t) [inline pubspec.yaml:87]: just curious can we remove `ios:` if nothing were set here 🙏

File: lib/main-production.dart [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import 'package:flutter/material.dart';
```
Comment (manh-t) [inline lib/main-production.dart:1]: I think the file name should follow snake_case convention, how do you think? 🙏

File: lib/main-staging.dart [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import 'package:flutter/material.dart';
```
Comment (manh-t) [inline lib/main-staging.dart:1]: same here

**Review by manh-t — COMMENTED** [review]
some minor issues

**Comment by manh-t** [comment]
@markgravity Could you please update `README.md` and add some important notes like how to run the project? 🙋  I would help devs/ reviewer easily launch your app without missing any required args 🙏

File: lib/main-production.dart (hunk 7 above)
Comment (markgravity) [reply lib/main-production.dart:1]: It's generated by flutter_flavorizr, so we can change it but have to this again if we re-generate them 😢

File: .run/Production.run.xml (hunk 5 above)
Comment (markgravity) [reply .run/Production.run.xml:7]: This file is generated by Android Studio, so I think we should keep it lol 🤔

File: lib/main-production.dart (hunk 7 above)
Comment (manh-t) [reply lib/main-production.dart:1]: oh 🧱

**Review by manh-t — APPROVED** [review]
rest LGTM

File: android/app/src/main/AndroidManifest.xml [hunk 9]
```diff
@@ -1,41 +1,24 @@
...
-            android:name=".MainActivity"
-            android:launchMode="singleTop"
-            android:theme="@style/LaunchTheme"
-            android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode"
-            android:hardwareAccelerated="true"
-            android:windowSoftInputMode="adjustResize">
-            <!-- Specifies an Android theme to apply to this Activity as soon as
+<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="com.example.survey">
+  <application android:label="@string/app_name" android:icon="@mipmap/ic_launcher">
+    <activity android:name=".MainActivity" android:launchMode="singleTop" android:theme="@style/LaunchTheme" android:configChanges="orientation|keyboardHidden|keyboard|screenSize|smallestScreenSize|locale|layoutDirection|fontScale|screenLayout|density|uiMode" android:hardwareAccelerated="true" android:windowSoftInputMode="adjustResize">
```
Comment (sleepylee) [inline android/app/src/main/AndroidManifest.xml:3]: your xml format is broken here...

File: ios/Runner/Info.plist [hunk 10]
```diff
@@ -1,45 +1,45 @@
...
+    <array>
+      <string>UIInterfaceOrientationPortrait</string>
+      <string>UIInterfaceOrientationPortraitUpsideDown</string>
+      <string>UIInterfaceOrientationLandscapeLeft</string>
+      <string>UIInterfaceOrientationLandscapeRight</string>
+    </array>
+    <key>UIViewControllerBasedStatusBarAppearance</key>
+    <false/>
+  </dict>
+</plist>
```
Comment (sleepylee) [inline ios/Runner/Info.plist:45]: there is a setting in AS to add EOF line, I think you should change it and seem like this has been formatted incorrectly 🙇

File: lib/main-production.dart (hunk 7 above)
Comment (sleepylee) [reply lib/main-production.dart:1]: we can think about it later when we create our own template. Under the hood flavorizr just runs some scripts to create boilerplate setup file for us, we can do this ourselves and remove flavorizr later and follow our correct convention. The current setup helps us move faster with the early learning step.

File: pubspec.yaml [hunk 11]
```diff
@@ -74,3 +75,28 @@ flutter:
...
+        applicationId: "dev.markg.survey"
+      ios:
+        bundleId: "dev.markg.survey"
+    staging:
+      app:
+        name: "Survey-Staging"
+      android:
+        applicationId: "dev.markg.survey.staging"
+      ios:
+        bundleId: "dev.markg.survey.staging"
```
Comment (sleepylee) [inline pubspec.yaml:102]: eof

**Review by sleepylee — COMMENTED** [review]
mostly you have the issue with EOF line. You may want to recheck as Android Studio has a setting that help you adding it automatically.

**Review by sleepylee — APPROVED** [review]

**Review by manh-t — APPROVED** [review]

**Pull request merged**